src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp

```diff
@@ -592,8 +592,16 @@ void GraphingCalculator::GraphSettingsButton_Click(Object ^ sender, RoutedEventA
 
 void GraphingCalculator::DisplayGraphSettings()
 {
-    m_graphSettings = ref new GraphingSettings();
-    m_graphSettings->ViewModel->GraphThemeSettingChanged += ref new EventHandler<bool>(this, &GraphingCalculator::OnGraphThemeSettingChanged);
+    if (m_graphSettingsVM == nullptr)
+    {
+        m_graphSettingsVM = ref new GraphingSettingsViewModel();
+        m_graphSettingsVM->IsAlwaysLightTheme = !IsMatchAppTheme;
+        m_graphSettingsVM->IsMatchAppTheme = IsMatchAppTheme;
+    }
+
+    m_graphSettingsVM->GraphThemeSettingChanged += ref new EventHandler<bool>(this, &GraphingCalculator::OnGraphThemeSettingChanged);
```

Comment:
You want to move this into the above If statement otherwise you will keep assigning this event multiple times and the code will run multiple times whenever it fires.